jaeger-core/src/test/java/com/uber/jaeger/samplers/PerOperationSamplerTest.java

```diff
@@ -0,0 +1,167 @@
+    when(defaultProbabilisticSampler.sample(operation, traceId)).thenReturn(SamplingStatus.of(true, new HashMap<String, Object>()));
+    when(defaultProbabilisticSampler.getSamplingRate()).thenReturn(DEFAULT_SAMPLING_PROBABILITY);
+  }
+
+  @After
+  public void tearDown() {
+    undertest.close();
+  }
+
+  @Test
+  public void testFallbackToDefaultProbabilisticSampler() {
+    undertest = new PerOperationSampler(0, operationToSamplers, defaultProbabilisticSampler, DEFAULT_LOWER_BOUND_TRACES_PER_SECOND);
+    SamplingStatus samplingStatus = undertest.sample(operation, traceId);
+    assertTrue(samplingStatus.isSampled());
+    verify(defaultProbabilisticSampler).sample(operation, traceId);
+  }
+
+  @Test
+  public void testCreateGuaranteedSamplerOnUnseenOperation() {
+    String newOperation = "new operation";
+    undertest.sample(newOperation, traceId);
+    assertEquals(new GuaranteedThroughputSampler(DEFAULT_SAMPLING_PROBABILITY,
+                                                 DEFAULT_LOWER_BOUND_TRACES_PER_SECOND),
+                 operationToSamplers.get(newOperation));
+  }
+
+  @Test
+  public void testPerOperationSamplerWithKnownOperation() {
+    GuaranteedThroughputSampler sampler = mock(GuaranteedThroughputSampler.class);
+    operationToSamplers.put(operation, sampler);
+
+    when(sampler.sample(operation, traceId)).thenReturn(SamplingStatus.of(true, new HashMap<String, Object>()));
+
+    SamplingStatus samplingStatus = undertest.sample(operation, traceId);
+    assertTrue(samplingStatus.isSampled());
+    verify(sampler).sample(operation, traceId);
+    verifyNoMoreInteractions(defaultProbabilisticSampler);
+  }
+
+  @Test
+  public void testUpdate(){
+    GuaranteedThroughputSampler guaranteedThroughputSampler = mock(GuaranteedThroughputSampler.class);
+    operationToSamplers.put(operation, guaranteedThroughputSampler);
+
+    PerOperationSamplingParameters perOperationSamplingParameters =
+        new PerOperationSamplingParameters(operation, new ProbabilisticSamplingStrategy(SAMPLING_RATE));
+    List<PerOperationSamplingParameters> parametersList = new ArrayList<>();
+    parametersList.add(perOperationSamplingParameters);
+
+    OperationSamplingParameters parameters =
+        new OperationSamplingParameters(DEFAULT_SAMPLING_PROBABILITY,
+                                        DEFAULT_LOWER_BOUND_TRACES_PER_SECOND, parametersList);
+
+    assertTrue(undertest.update(parameters));
+    //Checks that creating a new instance with the given parameters is the same as updating an existing instance
+    assertEquals(new PerOperationSampler(MAX_OPERATIONS, parameters), undertest);
+  }
+
+  @Test
+  public void testNoopUpdate(){
+    List<PerOperationSamplingParameters> parametersList = new ArrayList<>();
+    OperationSamplingParameters parameters = new OperationSamplingParameters(0, 0, parametersList);
```

Comment:
lastly, I would want this test to actually verify that per-op samplers did not change, instead of relying on the method claiming that via return param.